packages/devtools_app/lib/src/screens/memory/shared/heap/model.dart

```diff
@@ -43,12 +44,18 @@ class AdaptedHeapData {
     );
   }
 
-  factory AdaptedHeapData.fromHeapSnapshot(HeapSnapshotGraph graph) =>
-      AdaptedHeapData(
-        graph.objects
-            .map((e) => AdaptedHeapObject.fromHeapSnapshotObject(e))
-            .toList(),
-      );
+  static Future<AdaptedHeapData> fromHeapSnapshot(
+    HeapSnapshotGraph graph,
+  ) async {
+    final listOfFutures = graph.objects.map((e) async {
+      await delayForBatchProcessing();
```

Comment:
how many objects will there be? Do we really want to delay for every object or do we want to delay for every N objects? 

Here's an example from the CPU profiler where we calculate the batch size: https://github.com/kenzieschmoll/devtools/blob/scroll-to-time-range/packages/devtools_app/lib/src/screens/profiler/cpu_profile_transformer.dart#L52